### PR TITLE
Elevate system

### PR DESCRIPTION
**Pull request by claytonpbarrows**
This PR:
 - renames `System` to `_System`
 - renames `ConcreteSystem` to `System` and routes all construction of `System` through the construction of `_System`

### CODE REVIEW

File: src/base.jl [hunk 1]
```diff
@@ -104,55 +104,56 @@ function System(; buses=[Bus()],
...
+    A _System struct with a modified forecasts field
 """
-function add_forecast!(sys::System,fc::Pair{Symbol,Array{Forecast,1}})
+function add_forecast!(sys::_System,fc::Pair{Symbol,Array{Forecast,1}})
     sys.forecasts[fc.first] = fc.second
 end
 
-"""A System struct that stores all devices in arrays with concrete types.
+"""
+A _System struct that stores all devices in arrays with concrete types.
```
Comment (daniel-thom) [inline src/base.jl]: We probably shouldn't refer to _System.  You could copy its docstring here and then change _System to say that it is an internal implementation detail.

File: src/base.jl [hunk 2]
```diff
@@ -192,8 +193,59 @@ function ConcreteSystem(sys::System)
...
+                basepower::Float64,
+                forecasts::Union{Nothing, SystemForecasts},
+                services::Union{Nothing, Vector{ <: Service}},
+                annex::Union{Nothing,Dict}; kwargs...)
+
+    
+    _sys = _System(buses, generators, loads, branches, storage, basepower, forecasts, services, annex; kwargs...)
+    return System(_sys)
+end
+
```
Comment (daniel-thom) [inline src/base.jl]: I like your choice of constructors.  But didn't @jd-lara say that he wanted explicit constructors that dealt with optional Branches and Loads?

**Comment by daniel-thom** [comment]
Looks good to me except for the docstring.

**Comment by jd-lara** [comment]
Please do not merge this change just yet even if approved. I want to add the support for the ED and UC formulations before having to perform this update.

File: src/base.jl (hunk 1 above)
Comment (claytonpbarrows) [reply src/base.jl]: 073620e5840a3e2abc5179f3ac3e7499bc23ae9e resolves

File: src/base.jl (hunk 2 above)
Comment (claytonpbarrows) [reply src/base.jl]: 08abf44e91b479f54c265022309060ef23582340 resolves

**Pull request merged**